module/VuFind/src/VuFind/Form/Form.php

```diff
@@ -673,9 +681,27 @@ protected function parseConfig($formId, $config, $params)
                     $element['settings']['rows'] = 8;
                 }
             }
+
+            // Merge sender fields with any existing field definitions:
+            if ('name' === $element['name']) {
+                $element = array_merge($senderName, $element);
+                $senderName = null;
+            } elseif ('email' === $element['name']) {
+                $element = array_merge($senderEmail, $element);
```

Comment:
Should we add a test case to prove that this new logic works as expected?

Reply:
Yes. I'll add the tests.

Reply:
Thanks! I expect I'll be able to merge this whenever you have a chance to complete that.

Reply:
Added tests and fixed an issue with default field size handling being done too early and overriding the sender fields. Thanks for suggesting the tests! :)

Reply:
Great! Writing tests sometimes feels like a burden, but it pays off very frequently -- glad this was one of those cases.

One suggestion for an additional test case: what happens if you provide a custom email element with a settings array that does not include a size? Does the default size get merged in correctly, or do all the settings get overwritten? I _think_ the current code will work as desired, but the behavior of `array_merge` on nested arrays is not especially well-documented, so it might be good to capture our desired behavior in a test just to be sure things are doing what they're supposed to.

Reply:
Good point. It didn't work as it overrides the content of top-level elements. I switched to array_replace_recursive, added tests and also fixed the code to not override aria-label if it's set in the settings of a field.